calculate_gain2.py

```
@@ -0,0 +1,50 @@
+
+    # check if test file exists
+    rel = os.path.relpath(t, 'lib')
+    base = rel.replace('.dart', '_test.dart')
+    test_path = os.path.join('test', base)
+
+    name = os.path.basename(t).replace('.dart', '')
+    found = False
+    if os.path.exists(test_path):
+        found = True
+    else:
+        for r, _, fs in os.walk('test'):
```

Ruff (0.15.13)

[warning] Loop control variable `r` not used within loop body

Rename unused `r` to `_r`

(B007)